src/components/interaction-designer/toolbar/TreeBuilderToolbar.vue

```diff
@@ -53,7 +53,7 @@
                 ref="edit-flow-modal"
                 ok-only
                 :ok-title="'flow-builder.done' | trans"
-                @ok="showOrHideEditFlowModal">
+                @ok="handleUpdateFlowDetails">
```

Comment:
The flow is updated whether you click okay to close the modal or you just click outside the modal which means that this doesn't quite work - I can add a new mode and it won't validate all resource fields

Reply:
Good catch @jcbashdown !
After thinking about this, I'm persuaded we should not have the close button for now & should not allow users to close the modal clicking outside. In fact, we bind the active flow itself to the form.

Therefore, see the associated commit https://github.com/FLOIP/flow-builder/pull/239/commits/1311ddd77fafe083131c6bfb95c79d1bbe81c840

Let me know if you have any concerns.

Reply:
Hmm. Can you explain a bit more about what was wrong with the old approach to this? Why not hook into changes on those fields?

The approach you are taking could have some advantages but it's a significant enough change to the UX that you might need to clear this with design.